File: Part1_강의자료/Ch08_투포인터/문제별코드/13422_도둑/13422_prefix_sum.java
import java.util.Scanner;

class Main
{
    static int getRangeSum(int[] acc, int l, int r) {
        if (l <= r) return acc[r] - acc[l - 1];
        return acc[acc.length - 1] - acc[l - 1] + acc[r];
    }

    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        while (T-- > 0) {
            int N = sc.nextInt();
            int M = sc.nextInt();
            int K = sc.nextInt();

            int[] money = new int[N + 1];
            for (int i = 1; i <= N; i++)
                money[i] = sc.nextInt();

            int[] accMoney = new int[N + 1];
            for (int i = 1; i <= N; i++)
                accMoney[i] = accMoney[i - 1] + money[i];

            int ans = 0;
            for (int i = 1; i <= N; i++) {
                if (getRangeSum(accMoney, i, (i + M - 2) % N + 1) < K)
                    ans++;
            }
            if (N == M) ans = Math.min(1, ans);
            System.out.println(ans);
        }
    }
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/13422_도둑/13422_prefix_sum.py
def get_range_sum(acc, l, r):
    if l <= r:
        return acc[r] - acc[l - 1]
    return acc[len(acc) - 1] - acc[l - 1] + acc[r]

for _ in range(int(input())):
    N, M, K = map(int, input().split())
    money = [0] + list(map(int, input().split()))
    acc_money = [0] * (N + 1)
    for i in range(1, N + 1):
        acc_money[i] = acc_money[i - 1] + money[i]

    ans = 0
    for i in range(1, N + 1):
        if get_range_sum(acc_money, i, (i + M - 2) % N + 1) < K:
            ans += 1
    if N == M:
        ans = min(1, ans)
    print(ans)
